src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -390,14 +391,64 @@ export interface ISeriesItemConfig {
     xAxis?: number;
 }
 
+export interface IPatternOptionsObject {
+    path: Highcharts.SVGAttributes;
+    width: number;
+    height: number;
+}
+export interface IPatternObject {
+    pattern: IPatternOptionsObject;
+}
+
+export interface IDataPoint {
+    x: number;
+    y: number;
+    value: number;
+    borderWidth?: number;
+    borderColor?: string;
+    color?: string | IPatternObject;
+    pointPadding?: number;
+}
+
 export function getHeatmapSeries(
     executionResultData: Execution.DataValue[][],
     measureGroup: Execution.IMeasureGroupHeader["measureGroupHeader"],
 ) {
-    const data = [] as any;
-    executionResultData.forEach((rowItem: any, rowItemIndex: number) => {
-        rowItem.forEach((columnItem: any, columnItemIndex: number) => {
-            data.push({ x: columnItemIndex, y: rowItemIndex, value: parseValue(columnItem) });
+    const data: IDataPoint[] = [];
+    const nullColor: IPatternObject = {
+        pattern: {
+            path: {
+                d: "M 10 0 L 0 10 M 9 11 L 11 9 M 4 11 L 11 4 M -1 1 L 1 -1 M -1 6 L 6 -1",
+                stroke: GRAY,
+                strokeWidth: 1,
+                fill: WHITE,
+            },
+            width: 10,
+            height: 10,
+        },
+    };
+    let value: number;
+    let pointData: IDataPoint;
+    executionResultData.forEach((rowItem: Execution.DataValue[], rowItemIndex: number) => {
+        rowItem.forEach((columnItem: Execution.DataValue, columnItemIndex: number) => {
+            value = parseValue(columnItem + "");
+            pointData = { x: columnItemIndex, y: rowItemIndex, value };
+            if (isNil(value)) {
+                data.push({
+                    ...pointData,
+                    borderWidth: 1,
+                    borderColor: GRAY,
+                    color: "transparent",
```

Comment:
we can add `const TRANSPARENT = "transparent"` to `../utils/color` and re-use it here